src/task/src/runners/azurePipelinesRunnerInvoker.ts

```diff
@@ -50,18 +50,16 @@ export default class AzurePipelinesRunnerInvoker
     });
   }
 
-  public getInput(name: string[]): string | undefined {
+  public getInput(name: string[]): string | null {
     const formattedName: string = name.join("");
     return this._azurePipelinesRunnerWrapper.getInput(formattedName);
   }
 
-  public getEndpointAuthorization(
-    id: string,
-  ): EndpointAuthorization | undefined {
-    const result: taskLib.EndpointAuthorization | undefined =
+  public getEndpointAuthorization(id: string): EndpointAuthorization | null {
+    const result: taskLib.EndpointAuthorization | null =
       this._azurePipelinesRunnerWrapper.getEndpointAuthorization(id, true);
-    if (!result) {
-      return undefined;
+    if (result === null) {
```

Comment:
This looks like a change in behavior.
!result would be true for 0 or "" whereas result === null would be true for those cases

Reply:
I agree that it is subtly different, so thanks for calling this out. But I believe this should be unproblematic here as the result should never be an integer but a JSON object. After the change, the compiler complained about this, which is the reason I've changed this. I also believe the changes in the wrapper class here means that the `0` case is actually covered anyway, albeit in a different location.